### PR TITLE
feat: disable provider test ping

### PR DESCRIPTION
**Pull request by ganchoradkov**
## Description
- Added optional `disableProviderPing` flag to disable the `http-provider` `test` request.
- Reworked how globally used values are shared between the provider, sub providers & utils. With the growing number of params being used copy pasting repetitive values on each sub-provider feels suboptimal.

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
tests, 
dogfooding
## Fixes/Resolves (Optional)

Please list any issues that are fixed by this PR in the format `#<issue number>`. If it fixes multiple issues, please put each issue in a separate line. If this Pull Request does not fix any issues, please delete this section.

## Examples/Screenshots (Optional)

Please attach a screenshot or screen recording of features/fixes you've implemented to verify your changes. Please also note any relevant details for your configuration. If your changes do not affect the UI, please delete this section.

Use this table template to show examples of your changes:

| Before       | After        |
| ------------ | ------------ |
| `test` json rpc request | No request being sent - depending on the flag |

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

# Additional Information (Optional)

Please include any additional information that may be useful for the reviewer.

### CODE REVIEW

File: providers/universal-provider/src/UniversalProvider.ts [hunk 1]
```diff
@@ -262,6 +266,11 @@ export class UniversalProvider implements IUniversalProvider {
         Object.keys(this.session.namespaces).map((namespace) => parseNamespaceKey(namespace)),
       ),
     ];
+
+    setGlobal("client", this.client);
+    setGlobal("events", this.events);
```
Comment (ignaciosantise) [inline providers/universal-provider/src/UniversalProvider.ts:271]: do we need to check if `this.events` is not null? or it doesn't matter?

File: package-lock.json [hunk 2]
```diff
@@ -33442,6 +33458,13 @@
         "@esbuild/win32-x64": "0.17.0"
       }
     },
+    "esbuild-darwin-arm64": {
```
Comment (ignaciosantise) [inline package-lock.json]: is this ok?

**Review by ignaciosantise — APPROVED** [review]
Looks good 🚀  i left some comments

File: providers/universal-provider/src/UniversalProvider.ts (hunk 1 above)
Comment (ganchoradkov) [reply providers/universal-provider/src/UniversalProvider.ts:271]: `this.events` is assigned on initialization and the types are preventing it from being `undefined`

**Pull request merged**